### ISSUE TITLE
Nav block lacks context about menu currently being edited within the block

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#47020]
As reported in https://github.com/WordPress/gutenberg/issues/46939#issuecomment-1374652632 when entering the Nav block there is no context as to the menu were are editing. It should explain whether it's auto menu or one of the menus we've created. Also perhaps it should announce whether the menu is "saved" or in "draft"?

An accessible `Name` should be provided along the lines of  `Navigation Menu: %%NAV MENU NAME%%` (also consider when the menu is an "auto" menu). 

Related https://github.com/WordPress/gutenberg/issues/47019

**Comment by scruffian** [comment]
I'm not sure how we would do this. The block itself has an aria-label of "Block: Navigation", which is consistent with all other blocks. I'm hesitant to make an exception for this block.

In addition, if we create menus seamlessly (see [Navigation: Create first menu seamlessly](https://github.com/WordPress/gutenberg/issues/47058)) then this information become redundant.

**Comment by annezazu** [comment]
@alexstine this touches on a piece of feedback you had in your recent 6.3 video. Mind giving [this PR](https://github.com/WordPress/gutenberg/pull/53469) a review when you have a chance? cc @joedolson and @afercia in case you all can help too!

**Comment by alexstine** [comment]
Approved the change. Thanks. 👍
